doc/source/coding-style/pep8.rst

```diff
@@ -0,0 +1,813 @@
+                   ...
+        
+               else:
+        
+                   ...
+
+           if x > 0 and x < 10:
+        
+               print("x is a positive single digit.")
+    
+    .. tab:: Use
+    
+        .. code-block:: python
+        
+           if x < y:
+               ...
+           else:
+               if x > y:
+                   ...
+               else:
+                   ...
+        
+           if x > 0 and x < 10:
+               print("x is a positive single digit.")
+           elif x < 0:
+               print("x is less than zero.")
+
+
+Maximum Line Length
+-------------------
+For source code lines, best practice is to keep the length at or below
+100 characters. For docstrings and comments, best practice is to keep
+the length at or below 72 characters.
+
+Lines longer than these recommended limits might not display properly
+on some terminals and tools or might be difficult to follow. For example,
+this line is difficult to follow:
+
+
+.. tabs::
+
+    .. tab:: Avoid
+
+        .. code:: python
+
+            employee_hours = [schedule.earliest_hour for employee in self.public_employees for schedule in employee.schedules]
+
+    .. tab:: Use
+
+        .. code-block:: python
+        
+            employee_hours = [
+                schedule.earliest_hour
+                for employee in self.public_employees
+                for schedule in employee.schedules
+            ]
+
+Alternatively, instead of writing a list comprehension, you can use a
+classic loop.
+
+Notice that sometimes it will not be possible to keep the line length below the
+desired value without breaking the syntax rules.
+
+Comments
+--------
+Because a PyAnsys library generally involves multiple physics domains,
+users reading its source code do not have the same background as
+the developers who wrote it. This is why it is important for a library
+to have well commented and documented source code. Comments that
+contradict the code are worse than no comments. Always make a priority
+of keeping comments up to date with the code.
+
+Comments should be complete sentences. The first word should be
+capitalized, unless it is an identifier that begins with a lowercase
+letter.
+
+Here are general guidelines for writing comments:
+
+#. Always try to explain yourself in code by making it
+   self-documenting with clear variable names.
+#. Don't be redundant.
+#. Don't add obvious noise.
+#. Don't use closing brace comments.
+#. Don't comment out code that is unused. Remove it.
+#. Use explanations of intent.
+#. Clarify the code.
+#. Warn of consequences.
+
+Obvious portions of the source code should not be commented. 
+For example, the following comment is not needed:
+
+.. code:: python
+
+   # increment the counter
+   i += 1
+
+However, an important portion of the behavior that is not self-apparent
+should include a note from the developer writing it. Otherwise,
+future developers may remove what they see as unnecessary. 
```

Comment:
```suggestion
However, if code behavior is not self-apparent, it should be documented.
Otherwise, future developers might remove code that they see as unnecessary. 
```